### ISSUE TITLE
MySQL query error: Removing roles works but throws an exception after commit

### ISSUE THREAD

**Issue by bbonamin** [RolifyCommunity/rolify#96]
Rolify created a table in my database called `users_roles`

When calling the method `remove_role` on a user, it removes the role but after doing the commit, I think it's trying to show all the remaining roles with an invalid query:

``` ruby
pry(main)> user.roles
Role Load (0.3ms)  SELECT `roles`.* FROM `roles` INNER JOIN `users_roles` ON `roles`.`id` = `users_
roles`.`role_id` WHERE `users_roles`.`user_id` = 15
=> [#<Role id: 1, name: "elevated", resource_id: 1, resource_type: "Project", created_at: "2012-08-02
 14:44:52", updated_at: "2012-08-02 14:44:52">,
 #<Role id: 2, name: "super_admin", resource_id: nil, resource_type: nil, created_at: "2012-08-02 14:
45:39", updated_at: "2012-08-02 14:45:39">,
 #<Role id: 3, name: "elevated", resource_id: 2, resource_type: "Project", created_at: "2012-08-02 15
:23:56", updated_at: "2012-08-02 15:23:56">]

#######

[24] pry(main)> user.remove_role :elevated, Project.first
  Project Load (0.2ms)  SELECT `projects`.* FROM `projects` LIMIT 1
  Role Load (0.6ms)  SELECT `roles`.* FROM `roles` INNER JOIN `users_roles` ON `roles`.`id` = `users_
roles`.`role_id` WHERE `users_roles`.`user_id` = 15 AND `roles`.`name` = 'elevated' AND `roles`.`reso
urce_type` = 'Project' AND `roles`.`resource_id` = 1
   (0.1ms)  BEGIN
(0.8ms)  DELETE FROM `users_roles` WHERE `users_roles`.`user_id` = 15 AND `users_roles`.`role_id` 
IN (1)
   (0.3ms)  COMMIT
   (0.3ms)  SELECT COUNT(*) FROM `users` INNER JOIN `user_roles` ON `users`.`id` = `user_roles`.`user
_id` WHERE `user_roles`.`role_id` = 1
ActiveRecord::StatementInvalid: Mysql2::Error: Table 'automation_wa.user_roles' doesn't exist: SELECT
 COUNT(*) FROM `users` INNER JOIN `user_roles` ON `users`.`id` = `user_roles`.`user_id` WHERE `user_r
oles`.`role_id` = 1
from /Users/bcb/.rvm/gems/ruby-1.9.2-p320/gems/rack-mini-profiler-0.1.7/lib/patches/sql_patches.rb:45
:in `query'
```

TL;DR: Removing roles works but it throws an exception in the end.
